### PR TITLE
Stable prep

### PR DESCRIPTION
**Pull request by abea**
Preparing for stable release, this adds a punch list to the README. It also addresses template headings and adds example filters.

@bulldoguk I'd welcome your review as well if you'd like.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
-This bundle is intended to provide event functionality (concerts, recurring webinars, etc.) with <a href="https://apostrophecms.com/">Apostrophe CMS 3</a>. It is based on the `apostrophe-events` bundle for Apostrophe CMS 2. 
+This bundle is intended to provide event functionality (concerts, recurring webinars, etc.) with <a href="https://apostrophecms.com/">Apostrophe CMS 3</a>. It is based on the `apostrophe-events` bundle for Apostrophe CMS 2.
+
+## Stable release punch list
+- [ ] The index page should only show upcoming published events
+- [x] Index page default template should use H2 and H3 headings for sections (filters and events) and a paragraph or `date` for the date.
+- [ ] Index page should have pagination.
+- [ ] needs localization keys
+- [ ] Make event description optional
+- [ ] Change `groupId` to `eventGroupId`
```
Comment (abea) [inline README.md]: `groupId` is a little too generic. The property should be more specific to this module.

File: index.js [hunk 2]
```diff
@@ -179,12 +179,12 @@ module.exports = {
         }
       },
       afterPublish: {
-        async publishChildren(req, piece, options) {
+        async publishChildren(req, data) {
```
Comment (abea) [inline index.js:172]: Full docs for this event and its handler parameters: https://v3.docs.apostrophecms.org/reference/server-events.html#afterpublish

File: modules/@apostrophecms/event-page/views/filters.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        <a
+          style="{{ 'font-style: italic' if data.query.month == month.value }}"
+          href="{{ here(data.url, { month: month.value }) }}">{{ month.label }}</a>
+      </li>
+    {% endfor %}
+  </ul>
+
+  <h3>Day</h3>
+  <ul>
+    {% for day in data.filters.day %}
```
Comment (abea) [inline modules/@apostrophecms/event-page/views/filters.html:36]: We could use select fields for the sample. That might be better, but it'd require client-side JS.

File: modules/@apostrophecms/event-page/views/page.html [hunk 4]
```diff
@@ -1,11 +0,0 @@
-{% extends data.outerLayout %}
```
Comment (abea) [inline modules/@apostrophecms/event-page/views/page.html:1]: This was changed to index.html. page.html works to some degree, but I don't know if it includes all index page features.

File: modules/@apostrophecms/event-page/views/index.html [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+{% block main %}
+  <h2>Filters</h2>
+  {% render filters.render({
+    filters: data.piecesFilters,
+    query: data.query,
+    url: data.page._url
+  }) %}
+  <h2>Events</h2>
+  {% for piece in data.pieces %}
+    <article class="apos-event-preview">
```
Comment (abea) [inline modules/@apostrophecms/event-page/views/index.html]: I think we should really remove all these classes. We don't provide starter styles and I would not expect anyone to use these templates as-is. They're primarily starter files to replace in the project.

File: README.md [hunk 6]
```diff
@@ -1,4 +1,13 @@
 # @apostrophecms/event
 Note: This module is in Alpha and not ready for use. It is a community-supported bundle and contributions are welcome!
 
-This bundle is intended to provide event functionality (concerts, recurring webinars, etc.) with <a href="https://apostrophecms.com/">Apostrophe CMS 3</a>. It is based on the `apostrophe-events` bundle for Apostrophe CMS 2. 
+This bundle is intended to provide event functionality (concerts, recurring webinars, etc.) with <a href="https://apostrophecms.com/">Apostrophe CMS 3</a>. It is based on the `apostrophe-events` bundle for Apostrophe CMS 2.
+
+## Stable release punch list
+- [ ] The index page should only show upcoming published events
```
Comment (boutell) [inline README.md]: Unless filtered by year or month, at which point it shows all events whose span of time touches that year or month.

Comment (boutell) [reply README.md]: In the A2 code that was being ported, the upcoming filter (now query builder) defaults to true, and is cancelled automatically if you set the year or month filters.

File: modules/@apostrophecms/event-page/views/filters.html [hunk 7]
```diff
@@ -0,0 +1,40 @@
+{%- macro here(url, changes) -%}
+  {{ url | build({
+    year: data.query.year
```
Comment (boutell) [inline modules/@apostrophecms/event-page/views/filters.html]: Missing month which may also be present.

File: modules/@apostrophecms/event-page/views/index.html (hunk 5 above)
Comment (boutell) [reply modules/@apostrophecms/event-page/views/index.html]: Fine with removing the classes.

File: queries.js [hunk 8]
```diff
@@ -76,11 +85,17 @@ module.exports = (self, query) => {
             .upcoming(null)
             .toDistinct('startDate');
 
-          const years = [{ value: null, label: 'All' }];
+          const years = [ {
+            value: null,
+            label: 'All'
+          } ];
           for (const eachdate of alldates) {
```
Comment (boutell) [inline queries.js]: Not a big deal because they are local variables, but `eachDate` and `allDates` read better.

File: queries.js [hunk 9]
```diff
@@ -102,11 +117,12 @@ module.exports = (self, query) => {
           const re = new RegExp(`${month}-`, 'gi');
```
Comment (boutell) [inline queries.js]: Needs leading `^`, which may have been missing in the past.

File: queries.js [hunk 10]
```diff
@@ -144,11 +166,12 @@ module.exports = (self, query) => {
           const re = new RegExp(`${day}`, 'gi');
 
           query.and({
-            $and: [{ startDate: re }, { endDate: re }]
+            $and: [ { startDate: re }, { endDate: re } ]
```
Comment (boutell) [inline queries.js]: This approach doesn't catch a three day event that includes the day we're looking for in the middle. And our date prefixes and actual date fields are fully comparable as strings. So I don't get why this part was changed from the 2.x approach.

From the 2.x code:

```
      finalize: function() {
        var day = self.get('day');

        if (day === null) {
          return;
        }

        self.and({
          startDate: { $lte: day },
          endDate: { $gte: day }
        });
      },
```

We should go back to this approach for the year, month and day filters. It's simpler and more efficient than regular expressions and handles the "day in the middle" case correctly.

**Review by boutell — CHANGES_REQUESTED** [review]

File: queries.js (hunk 10 above)
Comment (abea) [reply queries.js]: I'll add this to the punch list since it wasn't an edit from this PR.

Comment (boutell) [reply queries.js]: Looks like you changed your mind here, thanks for that, regressions from 2.x in terms of correctness don't make sense.

**Review by boutell — APPROVED** [review]

File: queries.js (hunk 10 above)
Comment (abea) [reply queries.js]: Not sure what you mean here. That was from @bulldoguk 's original code.

**Pull request merged**